src/librustc_builtin_macros/asm.rs

```diff
@@ -264,27 +270,72 @@ fn parse_args<'a>(
     Ok(args)
 }
 
+fn warn_duplicate_option<'a>(p: &mut Parser<'a>, span: Span) {
+    let mut warn = if let Ok(snippet) = p.sess.source_map().span_to_snippet(span) {
+        p.sess
+            .span_diagnostic
+            .struct_span_warn(span, &format!("the `{}` option was already provided", snippet))
+    } else {
+        p.sess.span_diagnostic.struct_span_warn(span, "this option was already provided")
+    };
```

Comment:
We want an error on duplicate options, not a warning.

Reply:
Isn't that a breaking change, though?

Reply:
`asm!` is still unstable (and it is very new), so we're allowed to do that.

Reply:
Ah, I see.